### DISCUSSION TITLE
Topic eu-test metadata information unknown

### DISCUSSION THREAD

**Discussion by write2jaydeep** [edenhill/librdkafka#3833] [General]
Hi 
We have upgraded the library with the latest master branch 1.9.0. It stopped our working producer application. We are using Apache 2.13. Our application is running on Linux platform(Ubuntu)

G 152736 -0-KAFKA-INT: 1650896856.815 ARI-RDKAFKA-7-NOINFO: rdkafka#producer-1: [thrd:ssl://internal.com:9093/bo
                       otstrap]: Topic eu-axis-paymt-trn-auth-item-inbound metadata information unknown
G 152736 -0-KAFKA-INT: 1650896856.815 ARI-RDKAFKA-7-NOINFO: rdkafka#producer-1: [thrd:ssl://internal.com:9093/bo
                       otstrap]: Topic eu-axis-paymt-trn-auth-item-inbound partition count is zero: should refresh metadata
G 152736 -0-KAFKA-INT: 1650896856.815 ARI-RDKAFKA-7-NOINFO: rdkafka#producer-1: [thrd:ssl://internal.com:9093/bo
                       otstrap]: Topic eu-test metadata information unknown
G 152736 -0-KAFKA-INT: 1650896856.815 ARI-RDKAFKA-7-NOINFO: rdkafka#producer-1: [thrd:ssl://internal.com:9093/bo
                       otstrap]: Topic eu-test partition count is zero: should refresh metadata
G 152736 -0-KAFKA-INT: 1650896856.815 ARI-RDKAFKA-7-NOINFO: rdkafka#producer-1: [thrd:ssl://internal.com:9093/bo
                       otstrap]: Topic eu-test metadata information unknown

This error "Topic eu-test metadata information unknown" I have seen very first time.
does it mean the Apache kafka version is not compatible with a client?

**Comment by edenhill** [comment]
Please reproduce with debug=all and provide the logs, thanks .


tis 26 apr. 2022 kl. 15:44 skrev JD ***@***.***>:

**Comment by write2jaydeep** [comment]
Thank you @edenhill for your swift reply.
I have replaced DNS in the log file. please find the attached zip file.
[unknown_metadata_issue.zip](https://github.com/edenhill/librdkafka/files/8564196/unknown_metadata_issue.zip)
.

**Reply by edenhill** [reply]
The clock seems to be acting weirdly:
```
G 152735 -0-KAFKA-INT: 1650896855.047 ARI-RDKAFKA-7-CONNECT: rdkafka#producer-1: [thrd:main]: Not selecting any broker for cluster connection: still suppressed for 657972160088ms: no cluster connection
G 152735 -0-KAFKA-INT: 1650896855.047 ARI-RDKAFKA-7-CONNECT: rdkafka#producer-1: [thrd:main]: Not selecting any broker for cluster connection: still suppressed for 657972160088ms: no cluster connection
G 152735 -0-KAFKA-INT: 1650896855.047 ARI-RDKAFKA-7-CONNECT: rdkafka#producer-1: [thrd:main]: Not selecting any broker for cluster connection: still suppressed for 657972160088ms: no cluster connection
G 152735 -0-KAFKA-INT: 1650896855.047 ARI-RDKAFKA-7-CONNECT: rdkafka#producer-1: [thrd:main]: Not selecting any broker for cluster connection: still suppressed for 657972160088ms: no cluster connection
G 152735 -0-KAFKA-INT: 1650896855.047 ARI-RDKAFKA-7-CONNECT: rdkafka#producer-1: [thrd:main]: Not selecting any broker for cluster connection: still suppressed for 657972160088ms: no cluster connection
G 152735 -0-KAFKA-INT: 1650896855.047 ARI-RDKAFKA-7-CONNECT: rdkafka#producer-1: [thrd:main]: Not selecting any broker for cluster connection: still suppressed for 657972160088ms: no cluster connection
G 152735 -0-KAFKA-INT: 1650896855.047 ARI-RDKAFKA-7-CONNECT: rdkafka#producer-1: [thrd:main]: Not selecting any broker for cluster connection: still suppressed for 657972160088ms: no cluster connection

```

Can you provide some details of the os, arch, containerization, etc, this runs on?

**Reply by write2jaydeep** [reply]
OpenVOS Release 19.1.0av Wed Mar 13 13:52:21 UTC 2019 i786 VOS

It is a Linux-like POSIX supported platform.

**Reply by write2jaydeep** [reply]
It works well with older release 1.8.0

**Reply by edenhill** [reply]
OpenVOS is not a supported platform, so you will need to find the root cause of this yourself unfortunately.

I recommend that you use `git bisect`  between v1.8.0 and master to find the commit that broke the clock.

**Reply by write2jaydeep** [reply]
Thank you very much @edenhill 
Most of the time I got a solution from you, considering this platform as Linux Ubuntu.

I think this problem seems to be platform-specific. You are right, I need to find the solution myself.

Thank you for your great help!
